discord/app_commands/transformers.py

```diff
@@ -249,6 +256,34 @@ async def transform(cls, interaction: Interaction, value: Any) -> Any:
         """
         raise NotImplementedError('Derived classes need to implement this.')
 
+    @classmethod
+    @_not_overridden
+    async def autocomplete(cls, interaction: Interaction, value: ChoiceT) -> List[Choice[ChoiceT]]:
```

Comment:
In public interfaces for inherited members, the type hints get auto-filled in when using certain tools so these need to be publicly named types.
```suggestion
    async def autocomplete(cls, interaction: Interaction, value: Union[int, float, str]) -> List[Choice[Union[int, float, str]]]:
```